src/api/mutations/payment/usePostPaymentToStripeMutation.ts

```diff
@@ -0,0 +1,56 @@
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+import { useMutation } from '@tanstack/react-query';
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import httpClient from '../../httpClient';
+import { RoutesConfig } from '../../../config/routes.config';
+import { ResponsePayment } from '../../../types/payment.types';
+import { EndpointsConfig } from '../../../config/endpoints.config';
+import { LOCAL_STORAGE_KEYS } from '../../../config/local-storage.config';
+import { getValueFromLocalStorage, removeFromLocalStorage } from '../../../utils';
+
+function usePostPaymentToStripeMutation(bookingId: string) {
+  const elements = useElements();
+  const stripe = useStripe();
+  const navigate = useNavigate();
+
+  const { mutate: mutatePaymentConfirm } = useMutation({
+    mutationFn: async (status: string) => {
+      const { data } = await httpClient.post<ResponsePayment>(
+        EndpointsConfig.Payment.PostConfirmPayment,
+        {
+          bookingId,
+          status,
+        }
+      );
+      return data;
+    },
+    onSuccess: (data) => {
+      removeFromLocalStorage(LOCAL_STORAGE_KEYS.clientSecret);
+      toast.success(data.message);
+      navigate(RoutesConfig.Root);
+    },
+  });
+
+  return useMutation({
+    mutationFn: async () => {
+      if (!stripe || !elements) return;
```

Comment:
i think we need here throw error like 
if (!stripe || !elements) throw Error
and handle the error inside component showing message 'unavailable, try later'